server/src/com/cloud/storage/snapshot/SnapshotSchedulerImpl.java

```diff
@@ -220,6 +237,26 @@ private void checkStatusOfCurrentlyExecutingSnapshots() {
     }
 
     @DB
+    protected void deleteExpiredVMSnapshots() {
+        String displayTime = DateUtil.displayDateInTimezone(DateUtil.GMT_TIMEZONE, _currentTimestamp);
+        Date now = new Date();
+
+        List<VMSnapshotVO> vmSnapshots = _vmSnapshotDao.listAll();
+        for (VMSnapshotVO vmSnapshot : vmSnapshots) {
+            long accountId = vmSnapshot.getAccountId();
+            int expiration_interval_hours = VMSnapshotManager.VMSnapshotExpireInterval.valueIn(accountId);
+            if (expiration_interval_hours < 0 ) {
+                continue;
+            }
+            Date creationTime = vmSnapshot.getCreated();
+            long diffInHours = TimeUnit.MILLISECONDS.toHours(now.getTime() - creationTime.getTime());
+            if (diffInHours > expiration_interval_hours) {
```

Comment:
but `long diffHours` can be either 4 or 5, in our case 4th hour will be the time between 3 and 4. When it gets to 4 it's in the 5th hour already. I'm +1 on '>='